src/check-reserve-update-frequency/check-reserve-update-frequency.js

```diff
@@ -0,0 +1,175 @@
+const ethers = require('ethers');
+const BigNumber = require('bignumber.js');
+const {
+  Finding, FindingSeverity, FindingType, getJsonRpcUrl,
+} = require('forta-agent');
+
+// load required shared types
+const {
+  LendingPoolAddressesProvider: lendingPoolAddressesProvider,
+  ProtocolDataProvider: protocolDataProviderAddress,
+} = require('../../contract-addresses.json');
+const { abi: protocolDataProviderAbi } = require('../../interfaces/AaveProtocolDataProvider.json');
+const { abi: aaveOracleAbi } = require('../../interfaces/AaveOracle.json');
+const { abi: chainlinkAggregatorAbi } = require('../../interfaces/IChainlinkAggregator.json');
+const { abi: lendingPoolAddressesProviderAbi } = require('../../interfaces/ILendingPoolAddressesProvider.json');
+
+// time threshold over which we trigger alerts (24 hours = 86400 seconds)
+// this value comes from the Chainlink web interface for price feeds (mouseover Trigger parameters)
+//  'A new trusted answer is written when the off-chain price moves more than the deviation
+//   threshold or 86400 seconds have passed since the last answer was written on-chain.'
+const {
+  checkReserveUpdateFrequency,
+  aaveEverestId: AAVE_EVEREST_ID,
+} = require('../../agent-config.json');
+
+const ORACLE_AGE_THRESHOLD_SECONDS = checkReserveUpdateFrequency.oracleAgeThresholdSeconds;
+
+// set up the an ethers provider
+// use ethers.providers.getDefaultProvider() in lieu of ethers.providers.WebSocketProvider()
+// websockets are not supported in production
+// eslint-disable-next-line new-cap
+const jsonRpcProvider = new ethers.providers.getDefaultProvider(getJsonRpcUrl());
+
+// there are several reserve tokens in AAVE that do not use Chainlink price oracles
+// we will filter these out before we attempt to determine the age of the oracle data
+// Gemini Dollar uses ExtendedGusdPriceProxy source: 0xEc6f4Cd64d28Ef32507e2dc399948aAe9Bbedd7e
+// xSUSHI Token uses XSushiPriceAdapter source: 0x9b26214bEC078E68a394AaEbfbffF406Ce14893F
+// Wrapped Ether uses Black Hole source: 0x0000000000000000000000000000000000000000
+const TOKENS_WITHOUT_CHAINLINK_ABI = [
+  '0x056Fd409E1d7A124BD7017459dFEa2F387b6d5Cd', // Gemini Dollar
+  '0x8798249c2E607446EfB7Ad49eC89dD1865Ff4272', // xSUSHI Token
+  '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2', // Wrapped Ether
+];
+
+// helper function to create alerts
+function createAlert(reserveToken, oracleAge, priceSourceAddress) {
+  return Finding.fromObject({
+    name: `Stale AAVE Price Oracle Data for ${reserveToken.symbol}`,
+    description: `Token ${reserveToken.symbol} Price Oracle Age: ${oracleAge} seconds`,
+    alertId: 'AE-AAVE-PRICE-ORACLE-STALE',
+    severity: FindingSeverity.Medium,
+    type: FindingType.Suspicious,
```

Comment:
Would FindingType.Degraded be more suitable in this case?

Reply:
Good suggestion.  I switched it.